### PR TITLE
DEV: Run CI with windows-latest

### PR DESCRIPTION
**Pull request by MartinThoma**

### CODE REVIEW

File: .github/workflows/github-ci.yaml [hunk 1]
```diff
@@ -17,6 +17,39 @@ on:
...
+    runs-on: windows-latest
+    steps:
+    - name: Checkout Code
+      uses: actions/checkout@v4
+      with:
+        submodules: 'recursive'
+    - name: Setup Python (3.11+)
+      uses: actions/setup-python@v4
+      with:
+        python-version: 3.11  # latest stable python
```
Comment (stefan6419846) [inline .github/workflows/github-ci.yaml:31]: This comment is wrong since October: https://www.python.org/downloads/release/python-3120/

Comment (MartinThoma) [reply .github/workflows/github-ci.yaml:31]: Uh, interesting! I didn't realize 3.12 is already stable :tada:  Thank you :hugs:

File: tests/test_filters.py [hunk 2]
```diff
@@ -259,6 +260,10 @@ def test_issue_399():
     reader.pages[1].extract_text()
 
 
+@pytest.mark.skipif(
```
Comment (pubpub-zz) [inline tests/test_filters.py:263]: I disagree with this. I've prepared a PR to fix this test.
Actually currently all tests are passing

**Comment by pubpub-zz** [comment]
@MartinThoma 
You've been too quick...

File: tests/test_filters.py (hunk 2 above)
Comment (MartinThoma) [reply tests/test_filters.py:263]: It should have been xfail and the reason was phrases bad, but I did see your pr :-)

**Pull request merged**